### PR TITLE
[MDS-5716] Facility Operator page for project description

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
- add the facility operator page and database columns to save the data
- bring over Map & LeafletMap to common, make tsx, make it update pins when you type and fit to bounds of map
- SteppedForm too, to common (the old antd Form and Menu.Item were giving me console errors and now they are not)
   - only real change here was calling it .tsx and getting rid of Menu.Item (pretty trivial)
- make the other inputs on the form new inputs so that it's not confusing what's required or not
- added a labelSubtitle param
- made project summary form not submit objects only containing nulls which were making the BE quite upset

[MDS-5716](https://bcmines.atlassian.net/browse/MDS-5716)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

**Review by matbusby-fw — DISMISSED** [review]
🦆

File: migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
+-- add party for facility operator
+ALTER TABLE project_summary
+    ADD COLUMN IF NOT EXISTS facility_operator_guid UUID,
+    ADD COLUMN IF NOT EXISTS facility_type VARCHAR,
```
Comment (henryoforeh-dev) [inline migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql]: Default character length here?

File: migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ADD COLUMN IF NOT EXISTS facility_desc VARCHAR(4000),
+    ADD COLUMN IF NOT EXISTS facility_latitude NUMERIC(9,7),
+    ADD COLUMN IF NOT EXISTS facility_longitude NUMERIC(11,7),
+    ADD COLUMN IF NOT EXISTS facility_coords_source VARCHAR(3),
+    ADD COLUMN IF NOT EXISTS facility_coords_source_desc VARCHAR(4000),
+    ADD COLUMN IF NOT EXISTS facility_pid_pin_crown_file_no VARCHAR(100),
+    ADD COLUMN IF NOT EXISTS legal_land_desc VARCHAR(4000),
+    ADD COLUMN IF NOT EXISTS facility_lease_no VARCHAR,
```
Comment (henryoforeh-dev) [inline migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql]: Same as comment above.

File: migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    ADD COLUMN IF NOT EXISTS zoning BOOLEAN,
+    ADD COLUMN IF NOT EXISTS zoning_reason VARCHAR,
```
Comment (henryoforeh-dev) [inline migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql]: Same here too.

File: services/core-api/app/api/projects/project_summary/models/project_summary.py [hunk 4]
```diff
@@ -282,54 +342,33 @@ def update(self,
...
+        self.facility_coords_source_desc = facility_coords_source_desc
+        self.facility_pid_pin_crown_file_no = facility_pid_pin_crown_file_no
+        self.legal_land_desc = legal_land_desc
+        self.facility_lease_no = facility_lease_no
+        self.zoning = zoning
+        self.zoning_reason = zoning_reason
+
+        if facility_operator:
+            if not facility_operator['party_type_code']:
+                facility_operator["party_type_code"] = "PER"
```
Comment (henryoforeh-dev) [inline services/core-api/app/api/projects/project_summary/models/project_summary.py:368]: Do we need an enum for party_type_code? Please disregard this comment if we have not used enum in other places where same should apply.

Comment (taraepp) [reply services/core-api/app/api/projects/project_summary/models/project_summary.py:368]: This one only has two possible values, PER and ORG, so I dunno, enum seems overkill to me. Codebase is indeed littered with literals for this field.

File: migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql (hunk 1 above)
Comment (taraepp) [reply migrations/sql/V2024.02.15.10.40__project_summary_facility_operator.sql]: Yeah I agree. I'll hound UX/Rebecca on this.

**Review by henryoforeh-dev — DISMISSED** [review]
Nice work!

File: services/core-api/app/api/projects/project_summary/models/project_summary.py (hunk 4 above)
Comment (matbusby-fw) [reply services/core-api/app/api/projects/project_summary/models/project_summary.py:368]: There are a few instances of enums in the codebase (I remember one or two in notice_of_departure), but not adopted a lot yet.

**Review by matbusby-fw — APPROVED** [review]

**Review by henryoforeh-dev — APPROVED** [review]

**Pull request merged**